frame/scheduler/src/lib.rs

```diff
@@ -212,7 +263,7 @@ decl_module! {
 							maybe_id,
 							r.map(|_| ()).map_err(|e| e.error)
 						));
-						result = cumulative_weight;
+						total_weight = cumulative_weight;
```

Comment:
I'm not sure this is correct:
if `cumulative_weight > limit` and there is no HARD_DEADLINE scheduled then only the first scheduled is executed. so I think only the weight to this call should be added.